dingo/gw/likelihood.py

```diff
@@ -192,13 +192,29 @@ def log_likelihood(self, theta):
             )
 
         if self.time_marginalization:
-            return self._log_likelihood_time_marginalized(theta)
+            result = self._log_likelihood_time_marginalized(theta)
         elif self.phase_marginalization:
-            return self._log_likelihood_phase_marginalized(theta)
+            result = self._log_likelihood_phase_marginalized(theta)
         elif self.calibration_marginalization_kwargs:
-            return self._log_likelihood_calibration_marginalized(theta)
+            result = self._log_likelihood(theta)
         else:
-            return self._log_likelihood(theta)
+            result = self._log_likelihood(theta)
+
+        # We assume that the _log_likelihood* methods return either a 2-tuple (log_likelihood, supplement), or just a
+        # log_likelihood scalar. If just a scalar, but supplemental information was requested, return an empty dict.
+
+        if include_supplemental:
```

Comment:
Why would some of the log_likelihood functions not return the supplement? If they did not compute the SNRs (or other info) for some reason. But then, they could return an empty dict and this code could be simplified.